definitions/procedures/backup/snapshot/mount_pulp.rb

```diff
@@ -13,10 +13,12 @@ class MountPulp < MountBase
       def run
         skip if @skip
         with_spinner('Creating snapshot of Pulp') do |spinner|
+          feature(:pulp).mark_base_directory
```

Comment:
Having a block-versoin of this method would proabely feel a bit better, so that we don't forget to unmark the file later (and we should probably do that in `ensure` block). This way, we could actually check when calling `find_base_directory(mount_point)`, that we're actually inside the block (by using some instance variable inside the feature). I'm not sure this is the best explanation of my suggestion - we can discuss it offline in more details if needed.

Reply:
I was thinking of a block-version too so I'll update. About the check I'm not sure because the find is happening on the snapshot much later and the original is no longer neither marked nor inside the block.

Reply:
Can't we leverage the fact that both methods are defined on the same object?